### PR TITLE
Improves the HUD

### PR DESCRIPTION
**Pull request by ike709**
## About the PR 
Improves QoL by adding commonly used inventory slots to the HUD, and makes it generally less terrible by arranging them in a grid in the corner.

Also removes the combat mode button and makes it an action.

**Screenshots**

**Outdated:**
image

image

**Latest:**
image

**Outdated:**
https://streamable.com/t7vqgl

**Changelog**

:cl:
- tweak: Completely redid the HUD inventory.
- add: Added an on-HUD inventory, using the same hotkey.
- remove: Nuked the inventory popup.
- tweak: Replaced the inventory slot icons and the disarm icon.
- tweak: Combat mode is now an action, rather than its own button.
- tweak: The default actions (in order) are now: Combat Mode, Disarm, Scream.

### CODE REVIEW

**Comment by SweptWasTaken** [comment]
Looks bad. You should decrease the amount of hotbar sloths by two and put the combat mode and doll underneath that.

**Comment by SweptWasTaken** [comment]
Comparison photo of our current UI layout

image

**Comment by Peptide90** [comment]
Personally I prefer this stuff on the left where you usually expect things like belt etc to be.

**Comment by ike709** [comment]
Nuked the combat mode button and made it an action. PR description updated with new pictures.

I need a unique icon for the combat mode action @SweptWasTaken

**Comment by ike709** [comment]
Added a harm icon from the same DMI as disarm. PR desc updated.

**Comment by SweptWasTaken** [comment]
Combat mode being assigned to hotbar is good. Everything else is unneeded.

File: Content.Client/GameObjects/Components/HUD/Inventory/HumanInventoryInterfaceController.cs [hunk 1]
```diff
@@ -67,27 +72,40 @@ void AddButton(out ItemSlotButton variable, Slots slot, string textureName)
             AddButton(out _hudButtonPocket1, Slots.POCKET1, "pocket");
             AddButton(out _hudButtonPocket2, Slots.POCKET2, "pocket");
             AddButton(out _hudButtonBack, Slots.BACKPACK, "back");
+            AddButton(out _hudButtonOClothing, Slots.OUTERCLOTHING, "suit_storage");
```
Comment (SweptWasTaken) [inline Content.Client/GameObjects/Components/HUD/Inventory/HumanInventoryInterfaceController.cs:75]: Why did you use suit_storage here. We have suit.

**Review by SweptWasTaken — CHANGES_REQUESTED** [review]

File: Content.Client/GameObjects/Components/HUD/Inventory/HumanInventoryInterfaceController.cs (hunk 1 above)
Comment (ike709) [reply Content.Client/GameObjects/Components/HUD/Inventory/HumanInventoryInterfaceController.cs:75]: Fixed.

**Comment by ike709** [comment]
image
image

**Comment by PJB3005** [comment]
This just moves all the common inventory buttons further away from the center of the screen and, as such the mouse.

It also wastes valuable screen real estate (since inventory buttons are a small UI element that are easy to fit at the bottom of the screen in a row, instead of chunking them together in a grid like this).

MASSIVE 👎 here.

**Comment by SweptWasTaken** [comment]
> This just moves all the common inventory buttons further away from the center of the screen and, as such the mouse.
> 
> 
> 
> It also wastes valuable screen real estate (since inventory buttons are a small UI element that are easy to fit at the bottom of the screen in a row, instead of chunking them together in a grid like this).
> 
> 
> 
> MASSIVE 👎 here.

I was gonna draw lines from the center of the screen to the UI and compare distance but I didn't because I thought people would call me schizo.

**Comment by ike709** [comment]
@PJB3005 I'd put it on the left with the action bar if it wasn't so ridiculously massive.

**Comment by ike709** [comment]
Making this a draft until goonhud is finished.

**Comment by ike709** [comment]
Visually it's fine, code-wise... let's see how review goes.

File: Resources/Textures/Interface/Inventory/meta.json [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    "copyright": "Taken from https://github.com/goonstation/goonstation at commit e77d85d9c1d93aa32da7702737ceeac2b56738ac.",
+    "states": [
+        {
+            "name": "back",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (metalgearsloth) [inline Resources/Textures/Interface/Inventory/meta.json:17]: Don't need these if it's default anymore

File: Content.Server/Actions/CombatMode.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        [DataField("combatOff")] public string MessageOff { get; private set; } = "You exit combat mode.";
+
+        public bool DoToggleAction(ToggleActionEventArgs args)
+        {
+            if (!args.Performer.TryGetComponent(out CombatModeComponent? combatMode))
+            {
+                return false;
+            }
+
+            args.Performer.PopupMessage(args.ToggledOn ? MessageOn : MessageOff);
```
Comment (metalgearsloth) [inline Content.Server/Actions/CombatMode.cs:26]: Is this localised?

File: Content.Client/GameObjects/Components/HUD/Inventory/InventoryInterfaceController.cs [hunk 4]
```diff
@@ -30,23 +30,11 @@ public virtual void Initialize()
         public virtual void PlayerAttached()
         {
             GameHud.InventoryButtonVisible = true;
-            GameHud.InventoryButtonToggled = b =>
```
Comment (metalgearsloth) [inline Content.Client/GameObjects/Components/HUD/Inventory/InventoryInterfaceController.cs:33]: Shouldn't this class be nuked entirely or is it out of scope? e.g. stuff like the Window property definitely shouldn't exist anymore.

**Review by metalgearsloth — APPROVED** [review]
Code seems okay. I'd probably merge in https://github.com/space-wizards/space-station-14/pull/3513 first

Also not sure where ItemStatusPanels would go.

File: Resources/Textures/Interface/Inventory/meta.json (hunk 2 above)
Comment (ike709) [reply Resources/Textures/Interface/Inventory/meta.json:17]: https://github.com/space-wizards/RSI-editor/issues/8

File: Content.Client/GameObjects/Components/HUD/Inventory/InventoryInterfaceController.cs (hunk 4 above)
Comment (ike709) [reply Content.Client/GameObjects/Components/HUD/Inventory/InventoryInterfaceController.cs:33]: Definitely out of scope. A lot of it is entangled with the HUD inventory code and rewriting a bunch of inventory code in a PR to rearrange the HUD is definitely not something I am going to do.

**Comment by SweptWasTaken** [comment]
I still do not like how this looks. Of every part of our current UI, this was the least problem.

The popup hotbar is nice, but it should be using the old icon sprites.

**Comment by ike709** [comment]
There's no point caring when neither style matches the HUD style.

**Comment by ike709** [comment]
Plus unlike the old icons, these are all the same size.

**Comment by SweptWasTaken** [comment]
> 
> 
> There's no point caring when neither style matches the HUD style.

One looks more out of place than the other. Use the old icons. They're different sizes for a reason.

**Comment by ike709** [comment]
I remembered why I changed it. The old ones were different sizes for no reason.

**Comment by ike709** [comment]
image

uSe ThE oLd OnEs

**Comment by SweptWasTaken** [comment]
Still looks bad. Instead of 2 conflicting UI designs it's 3.

**Comment by SweptWasTaken** [comment]
Stop being a baby, unblock me on discord then send me the icons in a zip and I'll fix them for you.

**Comment by ike709** [comment]
All-new sprites:
image

**Comment by ike709** [comment]
Build failures seem to be unrelated.

**Comment by SweptWasTaken** [comment]
What does it look like if the mob has multiple hands?

**Comment by ike709** [comment]
image

Old pic but still accurate. Just puts it between the slots, as before.

**Comment by SweptWasTaken** [comment]
Kk i'm gonna re-add the blank hand icon without the L or R.

**Comment by SweptWasTaken** [comment]
https://discord.com/channels/310555209753690112/770682801607278632/824776596786774033

Here is the icons. I've:

- Relayered the icons so that the actual small clothing sprite is on top of the scan-lines
- Added some ones you didn't include
- Changed the hue to fit more with the UI
- Changed the hat and mask icon to be more recognizable

If you could try them in-game and put photos here.

**Pull request merged**